### PR TITLE
Using Modern CSS Features in Old Browsers using @supports

### LINKED ISSUE #2066: Using Modern CSS Features in Old Browser using @supports
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article

Using Modern CSS Features in Old Browsers using @supports

### Introduction paragraph (2-3 paragraphs):

In the world of Web Development, there are a lot of web browsers like Chrome, Safari, Firefox, etc. They vary slightly in their support for CSS features and properties. To solve this, the `@supports` CSS rule was introduced. This article is going to give readers an introduction to the @supports rule and give a hands on about the different ways this rule can be used to use modern features without breaking support in old web browsers. 

### Key takeaways: 

- The reader will be exposed to the different types of web browsers and web engines.
- The reader will be given an introduction to the `@supports` CSS rule.
- The reader will be walked through the different ways to use modern features without breaking compatibility in old web browsers.

### References: 

https://developer.mozilla.org/en-US/docs/Web/CSS/@supports

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by cyberShaw**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2066

### CODE REVIEW

File: content/articles/css-supports-query/index.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  .text {
+    /* Sets the font size */
+    font-size: x-large;
+  }
+</style>
+```
+
+If you run it on Firefox 22, you would notice that the hero banner sticks to the top, even when we scroll to the bottom of the page.
+
+In the above example, we use the `@supports` feature query and the **not** operator to check for the browsers that do not support the `position: sticky` CSS rule. We specify some alternative CSS rules for older browsers within the `@supports` block so that the CSS doesn't break in older browsers.
```
Comment (ahmadmardeni1) [inline content/articles/css-supports-query/index.md:227]: You only talked about the "not" operation. Please talk about "or" and "and" operations. Create 3 headers and describe the differences between them and when to use them.

**Review by ahmadmardeni1 — CHANGES_REQUESTED** [review]
This is a great tutorial, good job. @cyberShaw 
Please do the suggestions I made.

**Comment by cyberShaw** [comment]
Thank you for the review!
I'll make the changes by tomorrow, @ahmadmardeni1

**Comment by ahmadmardeni1** [comment]
Hello 👋 @cyberShaw 
As you know, we were working on a new structure for our EngEd Repo, which has led to a slight slow down of article publishing. Everything is almost ready now.

In order to publish your article as soon as possible please do the following:
- Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
- Close this PR.
- Open new PR in our new folder. You can find the *article* folder inside *content*.
- Mention me in a comment to let the other peer reviewers know that am the one responsible for your PR.

Ping me via Slack if you have any problems.

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Comment by hectorkambow** [comment]
[Using Modern CSS Features in Old Browsers using @supports #2317.pdf](https://github.com/section-engineering-education/engineering-education/files/6563580/Using.Modern.CSS.Features.in.Old.Browsers.using.%40supports.2317.pdf)

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was way higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by cyberShaw** [comment]
> [Using Modern CSS Features in Old Browsers using @supports #2317.pdf](https://github.com/section-engineering-education/engineering-education/files/6563580/Using.Modern.CSS.Features.in.Old.Browsers.using.%40supports.2317.pdf)
> 
> Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was way higher than we typically accept.
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> Be sure to see our resources page to see more info on plagiarism and what is considered as such.

I'll work on fixing the flags, @hectorkambow

**Comment by cyberShaw** [comment]
Hey @ahmadmardeni1!
Thank you for reviewing my article :D

As you suggested, I've made some changes:

- Discussed all three scenarios of browser support, as suggested.
- Mentioned and explained (with examples) about the **and** and **or** operator.

Please check it out 😅

**Review by louisefindlay23 — APPROVED** [review]
Final review complete. Congrats on your published article, @cyberShaw.

**Comment by hectorkambow** [comment]
@cyberShaw quick question - your PR shows that it has two .md files? Did you mean to include these two ???

cc @louisefindlay23

**Comment by cyberShaw** [comment]
> @cyberShaw quick question - your PR shows that it has two .md files? Did you mean to include these two ???
> 
> cc @louisefindlay23 

No, I don't remember adding two .md files 🤔

**Comment by hectorkambow** [comment]
@cyberShaw thank you - Im looking through the commits - you're right it was added later. 👍

**Pull request merged**